src/test/plugins/mvcc_delete_plugin_system_test.cpp

```diff
@@ -0,0 +1,120 @@
+
+    std::vector<int> vec(MAX_CHUNK_SIZE);
+    std::iota(vec.begin(), vec.end(), 0);
+
+    Segments segments;
+    const auto value_segment = std::make_shared<ValueSegment<int>>(std::move(vec));
+    segments.emplace_back(value_segment);
+    auto table = std::make_shared<Table>(column_definitions, TableType::Data, MAX_CHUNK_SIZE, UseMvcc::Yes);
+    table->append_chunk(segments);
+
+    StorageManager::get().add_table("mvcc_test", table);
+  }  // managed by each test individually
+
+  void TearDown() override {
+    StorageManager::get().drop_table("mvcc_test");
+    StorageManager::reset();
+  }
+
+ protected:
+  // Update all values in the table to create invalidated rows.
+  void update_table() {
+    auto column = expression_functional::pqp_column_(ColumnID{0}, DataType::Int, false, "number");
+
+    auto& tm = TransactionManager::get();
+
+    const auto transaction_context = tm.new_transaction_context();
+    const auto value = static_cast<int>(counter++ % (MAX_CHUNK_SIZE));
+    const auto expr = expression_functional::equals_(column, value);
+
+    const auto gt = std::make_shared<GetTable>("mvcc_test");
+    gt->set_transaction_context(transaction_context);
+
+    const auto validate = std::make_shared<Validate>(gt);
+    validate->set_transaction_context(transaction_context);
+
+    const auto where = std::make_shared<TableScan>(validate, expr);
+    where->set_transaction_context(transaction_context);
+
+    const auto update = std::make_shared<Update>("mvcc_test", where, where);
+    update->set_transaction_context(transaction_context);
+
+    gt->execute();
+    validate->execute();
+    where->execute();
+    update->execute();
+
+    if (!update->execute_failed()) {
+      transaction_context->commit();
+    } else {
+      transaction_context->rollback();
+      counter--;
+    }
+  }
+
+  constexpr static size_t MAX_CHUNK_SIZE = 200;
+  constexpr static size_t PHYSICALLY_DELETED_CHUNKS_COUNT = 3;
+  size_t deleted_chunks = 0;
+  uint64_t counter = 0;
+};
+
+/**
+ * This test checks the number of physically deleted chunks, which means
+ * nullptrs in the table.
+ * These nullptrs are created when the plugin successfully deleted a chunk.
+ */
+TEST_F(MvccDeletePluginSystemTest, CheckPlugin) {
+  auto& pm = PluginManager::get();
+  pm.load_plugin(build_dylib_path("libMvccDeletePlugin"));
+
+  // This thread calls update_table() continuously.
+  std::unique_ptr<PausableLoopThread> table_update_thread =
+      std::make_unique<PausableLoopThread>(std::chrono::milliseconds(0), [&](size_t) { update_table(); });
+
+  while (deleted_chunks < PHYSICALLY_DELETED_CHUNKS_COUNT) {
+    const auto table = StorageManager::get().get_table("mvcc_test");
+    deleted_chunks = 0;
+
+    for (auto chunk_id = ChunkID{0}; chunk_id < table->chunk_count(); ++chunk_id) {
+      if (!table->get_chunk(chunk_id)) {
+        deleted_chunks++;
+      }
+    }
+    std::this_thread::sleep_for(std::chrono::milliseconds(1000));
+  }
+
```

Comment:
Do I understand you correctly that this is still WIP? Because right now, you're not testing anything.

Reply:
This test only ends if the plugin deleted four chunks successfully, otherwise it stays in the while loop. We will add update and time constraints so the test will always stop.